examples/window-app/telink/include/WindowCovering.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "LEDWidget.h"
+#include "PWMDevice.h"
+
+#include <app/clusters/window-covering-server/window-covering-server.h>
+
+#include <cstdint>
+
+using namespace chip::app::Clusters::WindowCovering;
+
+class WindowCovering
+{
+public:
+    enum class MoveType : uint8_t
+    {
+        LIFT = 0,
+        TILT,
+        NONE
+    };
+
+    struct AttributeUpdateData
+    {
+        chip::EndpointId mEndpoint;
+        chip::AttributeId mAttributeId;
+    };
+
+    WindowCovering();
+    static WindowCovering & Instance()
+    {
+        static WindowCovering sInstance;
+        return sInstance;
+    }
+
+    PWMDevice & GetLiftIndicator() { return mLiftIndicator; }
+    PWMDevice & GetTiltIndicator() { return mTiltIndicator; }
+
+    void StartMove(MoveType aMoveType);
+    void SetSingleStepTarget(OperationalState aDirection);
+    void SetMoveType(MoveType aMoveType) { mCurrentUIMoveType = aMoveType; }
+    MoveType GetMoveType() { return mCurrentUIMoveType; }
+    void PositionLEDUpdate(MoveType aMoveType);
+
+    static void SchedulePostAttributeChange(chip::EndpointId aEndpoint, chip::AttributeId aAttributeId);
+    static constexpr chip::EndpointId Endpoint() { return 1; };
+
+private:
+    void SetBrightness(MoveType aMoveType, uint16_t aPosition);
+    void SetTargetPosition(OperationalState aDirection, chip::Percent100ths aPosition);
+    uint8_t PositionToBrightness(uint16_t aPosition, MoveType aMoveType);
+
+    static void UpdateOperationalStatus(MoveType aMoveType, OperationalState aDirection);
+    static bool TargetCompleted(MoveType aMoveType, NPercent100ths aCurrent, NPercent100ths aTarget);
+    static void StartTimer(MoveType aMoveType, uint32_t aTimeoutMs);
+    static chip::Percent100ths CalculateSingleStep(MoveType aMoveType);
+    static void DriveCurrentLiftPosition(intptr_t);
```

Comment:
Ultimate goal would be to have all app demos using as much as possible of common code
i know this would need some refactor touching other vendors boards

But this would have been better than fork basically doing the same things

Reply:
Good suggestion, but this is a basic demo of the app, so we'd like to consider refactoring it with common files usage a bit sooner